security-command-center/snippets/src/main/java/management/api/CreateEventThreatDetectionCustomModule.java

```diff
@@ -0,0 +1,103 @@
+
+  public static void main(String[] args) throws IOException {
+    // https://cloud.google.com/security-command-center/docs/reference/security-center-management/rest/v1/organizations.locations.eventThreatDetectionCustomModules/create
+    // TODO: Developer should replace project_id with a real project ID before running this code
+    String projectId = "project_id";
+
+    String customModuleDisplayName = "custom_module_display_name";
+
+    createEventThreatDetectionCustomModule(projectId, customModuleDisplayName);
+  }
+
+  public static EventThreatDetectionCustomModule createEventThreatDetectionCustomModule(
+      String projectId, String customModuleDisplayName) throws IOException {
+
+    // Initialize client that will be used to send requests. This client only needs
+    // to be created
+    // once, and can be reused for multiple requests.
+    try (SecurityCenterManagementClient client = SecurityCenterManagementClient.create()) {
+
+      String name =
+          String.format(
+              "projects/%s/locations/global/eventThreatDetectionCustomModules/%s",
+              projectId, "custom_module");
+
+      // define the metadata and other config parameters severity, description,
+      // recommendation below
+      Struct metadataStruct =
+          Struct.newBuilder()
+              .putFields("severity", Value.newBuilder().setStringValue("MEDIUM").build())
+              .putFields(
+                  "description",
+                  Value.newBuilder().setStringValue("add your description here").build())
+              .putFields(
+                  "recommendation",
+                  Value.newBuilder().setStringValue("add your recommendation here").build())
+              .build();
+      Struct configStruct =
+          Struct.newBuilder()
+              .putFields("metadata", Value.newBuilder().setStructValue(metadataStruct).build())
+              .putFields(
+                  "ips",
+                  Value.newBuilder()
+                      .setListValue(
+                          ListValue.newBuilder()
+                              .addValues(Value.newBuilder().setStringValue("0.0.0.0").build())
+                              .build())
+                      .build())
+              .build();
+
+      // define the Event Threat Detection custom module configuration, update the EnablementState
+      // below
+      EventThreatDetectionCustomModule eventThreatDetectionCustomModule =
+          EventThreatDetectionCustomModule.newBuilder()
+              .setName(name)
+              .setConfig(configStruct)
+              .setDisplayName(customModuleDisplayName)
+              .setEnablementState(EnablementState.ENABLED)
+              .setType("CONFIGURABLE_BAD_IP")
+              .build();
+
+      CreateEventThreatDetectionCustomModuleRequest request =
+          CreateEventThreatDetectionCustomModuleRequest.newBuilder()
+              .setParent(String.format("projects/%s/locations/global", projectId))
```

Comment:
nit: is it possible that parent is not required since it duplicates information placed in the name? is it possible to define the name of the resource to use one project ID and location but to set parent to another project and location?

Reply:
Even I have also noticed that we are passing duplicate information in both parent and name. Actually I tried to remove the parent since we are passing the duplicate information but I am getting an exception, seems it will be a required field to pass in the request

Reply:
Addressed, now only parent is used